scvi/model/_scvi.py

```diff
@@ -188,4 +203,72 @@ def setup_anndata(
             fields=anndata_fields, setup_method_args=setup_method_args
         )
         adata_manager.register_fields(adata, **kwargs)
+        # register new fields for latent mode if needed
+        latent_mode = _get_latent_adata_type(adata)
+        if latent_mode is not None:
+            adata_manager.register_new_fields(cls._get_latent_fields(latent_mode))
         cls.register_manager(adata_manager)
+
+    def _get_latent_adata_from_adata(
+        self,
+        mode: LatentDataType,
+        use_latent_key: str = "X_latent",
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ) -> AnnData:
+        all_zeros = csr_matrix(self.adata.X.shape, dtype=np.int8).toarray()
+        bdata = AnnData(
+            X=all_zeros.copy(),
+            layers={layer: all_zeros.copy() for layer in self.adata.layers},
+        )
+        bdata.obs = self.adata.obs
+        bdata.obsm = self.adata.obsm
+        bdata.obsp = self.adata.obsp
+        bdata.uns = self.adata.uns
+        bdata.var_names = self.adata.var_names
+        # add latent related metadata
+        if mode == "sampled":
+            bdata.obsm[_SCVI_LATENT_SAMPLES] = self.adata.obsm[use_latent_key]
```

Comment:
I will not comment this everywhere, but I don't see the purpose of having multiple modes, most of the code added assumes parameterized posteriors, so we should always assume dist mode. This will make the code a bit simpler as well.

Reply:
I will remove it the next pr (i.e. after this one is merged in)